### ISSUE TITLE
Validate lineTotal

### ISSUE THREAD

**Issue by christianschmid200272** [stephanstapel/ZUGFeRD-csharp#488]
Hi,

actual I test the InvoiceValidator, the calculation of lineTotal is current based on NetUnitPrice *  BilledQuantity but I think right one is NetUnitPrice *  (BilledQuantity / UnitQuantity).

example:
unit 1000 pcs. 
price per unit 150€
purchased quantity 350

[https://github.com/stephanstapel/ZUGFeRD-csharp/blob/fa17b6425470ade812802cb95fab1209feab7830/ZUGFeRD/InvoiceValidator.cs#L73](url)

**Comment by stephanstapel** [comment]
Can you please include references to documentation and/ or schema?

**Comment by tobitege** [comment]
@christianschmid200272 The `UnitQuantity` is just an informational value and only available in the Extended profile.
It is not used in any schema files for price-related validations.

**Comment by christianschmid200272** [comment]
Entschuldigen Sie bitte meine späte Antwort, unsere Firewall hatte die Emails von GitHub in die Quarantäne verschoben.

Verwendet man Profil Extended mit UnitQuantity wird "lineTotal" aus meiner Sicht falsch ermittelt, da sich der Preis dann ja auf die UnitQuantity bezieht auch wenn es in keinem Schema zur Preisvalidierung enthalten ist. Es werden nicht Einheiten berechnet sondern eine Anzahl dessen Preis in einer Einheit mit einer definierten Menge angegeben ist.

Einheit 1000 Stück
Preis 150 €
Menge 350 Stück

Aktuelle Berechnung 
150 * 350 = 52500 
` total = decimal.Multiply(item.NetUnitPrice.Value, item.BilledQuantity); `

Alternative Berechnung die immer richtig sein sollte, egal welches Profil verwendet wird.
150 * (350 / 1000) = 22,5
` total = decimal.Multiply(item.NetUnitPrice.Value, System.Math.Max(1, decimal.Divide(item.BilledQuantity, (decimal)item.UnitQuantity))); `

**Comment by tobitege** [comment]
Ja, stimmt. Das macht durchaus Sinn, gutes Beispiel für Gebindegröße.
Aber ausserhalb von Extended gibt es halt den Wert nicht für UnitQuantity (hier: Einheit 1000) und der Empfänger hat somit nur 150 und 350 und würde Einheit 1 annehmen.

**Comment by christianschmid200272** [comment]
Ja, müsste man, so wie in meinem Beispiel gemacht. Die Alternative währe bei der Verwendung von Extended die UnitQuantity auf > 0 zu prüfen und dann die Berechnung entsprechend durchzuführen was das Verhalten klarer darstellen würde aber natürlich mehr Code bedeutet.
